model/map-file/src/main/java/org/keycloak/models/map/storage/file/FileMapKeycloakTransaction.java

```diff
@@ -148,6 +157,26 @@ protected boolean removeIfExists(Path sp) {
         protected String getTxId() {
             return tx.txId;
         }
+    }
+
+    private class IdProtector extends EntityFieldDelegate.WithEntity<V> {
+
+        public IdProtector(V entity) {
+            super(entity);
+        }
 
+        @Override
+        public <T, EF extends java.lang.Enum<? extends org.keycloak.models.map.common.EntityField<V>> & org.keycloak.models.map.common.EntityField<V>> void set(EF field, T value) {
+            String id = entity.getId();
```

Comment:
Should we also use `determineKeyFromValue` on this line?

Reply:
It could be used here, but would lead to the same value as `getId()`. And since `getId()` obtains a field value, it is faster.